components/proxy/src/main/kotlin/com/hotels/styx/routing/db/StyxObjectStore.kt

```diff
@@ -82,13 +93,18 @@ class StyxObjectStore<T> : ObjectStore<T> {
      * @property key object name
      * @property payload the object itself
      */
-    fun remove(key: T) {
-        queue {
-            val nextVersion = objects().minus(key)
+    fun remove(key: String) {
+        var current = objects.get()
+        var new = current.minus(key)
 
-            if (nextVersion != objects()) {
-                objects.set(nextVersion)
-                notifyWatchers(nextVersion)
+        while (!objects.compareAndSet(current, new)) {
+            current = objects.get()
+            new = current.minus(key)
+        }
+
+        queue {
+            if (current != new) {
```

Comment:
Could we check this condition before deciding to queue a task?

Reply:
Yes I think that should be fine.